logic/src/commonMain/kotlin/com/wire/kalium/logic/data/auth/DomainRegistrationMapper.kt

```diff
@@ -0,0 +1,49 @@
+/*
+ * Wire
+ * Copyright (C) 2025 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ */
+package com.wire.kalium.logic.data.auth
+
+import com.wire.kalium.network.api.unauthenticated.domainregistration.DomainRedirect
+import com.wire.kalium.network.api.unauthenticated.domainregistration.DomainRegistrationDTO
+
+interface DomainRegistrationMapper {
+    fun fromApiModel(domainRegistrationDTO: DomainRegistrationDTO, email: String): LoginDomainPath
+}
+
+internal object DomainRegistrationMapperImpl : DomainRegistrationMapper {
+    override fun fromApiModel(domainRegistrationDTO: DomainRegistrationDTO, email: String): LoginDomainPath {
+        return when (domainRegistrationDTO.domainRedirect) {
+            DomainRedirect.PRE_AUTHORIZED -> LoginDomainPath.Default
+            DomainRedirect.LOCKED -> LoginDomainPath.Default
```

Comment:
I lack domain knowledge on this one, so I might ask questions just for the sake of sharing.

What are `PRE_AUTHORIZED` and `LOCKED` / `locked values`?

I think I got all the rest.

Reply:
This was got lost in translation. BE was supposed to return only a handful of the full `DomainRedirectTag` from their Enum:

```
[ none, locked, sso, backend, no-registration, pre-authorized ]
```

Because they didn't, we are narrowing down, or just mapping to the fewer cases there are for the users.

What are these `PRE_AUTHORIZED` and `LOCKED`  values? Just backoffice/team configurations for the team.